diff -ruN '--exclude=.git' '--exclude=*.o' '--exclude=*.rej' '--exclude=*.orig' a/src/macros.h b/src/macros.h
--- a/src/macros.h	2026-09-21 13:40:41.000000000 -0700
+++ b/src/macros.h	2026-09-23 21:24:15.715044390 -0700
@@ -169,7 +169,9 @@
 # define mch_stat(n, p)		stat(vms_fixfilename(n), (p))
 #else
 # if !defined(MSWIN) && !defined(PROTO)
+#ifndef mch_access	// ESP port may override: see port/auto/config.h
 #  define mch_access(n, p)	access((n), (p))
+#endif
 # endif
 
 // Use 64-bit fstat function on MS-Windows.
@@ -186,7 +188,9 @@
 #  ifdef STAT_IGNORES_SLASH
 #   define mch_stat(n, p)	vim_stat((n), (p))
 #  else
+#ifndef mch_stat	// ESP port may override: see port/auto/config.h
 #   define mch_stat(n, p)	stat((n), (p))
+#endif
 #  endif
 # endif
 #endif
diff -ruN '--exclude=.git' '--exclude=*.o' '--exclude=*.rej' '--exclude=*.orig' a/src/os_unix.h b/src/os_unix.h
--- a/src/os_unix.h	2026-09-21 13:40:41.000000000 -0700
+++ b/src/os_unix.h	2026-09-23 21:24:15.715590280 -0700
@@ -62,8 +62,12 @@
 #  define mch_remove(x) delete((char *)vms_fixfilename(x))
 # else
 #  define vim_mkdir(x, y) mkdir((char *)(x), y)
+#ifndef mch_rmdir	// ESP port may override: see port/auto/config.h
 #  define mch_rmdir(x) rmdir((char *)(x))
+#endif
+#ifndef mch_remove	// ESP port may override: see port/auto/config.h
 #  define mch_remove(x) unlink((char *)(x))
+#endif
 # endif
 #endif
 
diff -ruN '--exclude=.git' '--exclude=*.o' '--exclude=*.rej' '--exclude=*.orig' a/src/vim.h b/src/vim.h
--- a/src/vim.h	2026-09-21 13:40:41.000000000 -0700
+++ b/src/vim.h	2026-09-23 21:24:15.714355129 -0700
@@ -2579,9 +2579,13 @@
 // This must come after including proto.h.
 // For VMS this is defined in macros.h.
 #if !defined(MSWIN) && !defined(VMS) && !defined(PROTO)
+#ifndef mch_open	// ESP port may override: see port/auto/config.h
 # define mch_open(n, m, p)	open((n), (m), (p))
+#endif
+#ifndef mch_fopen	// ESP port may override: see port/auto/config.h
 # define mch_fopen(n, p)	fopen((n), (p))
 #endif
+#endif
 
 #include "globals.h"	    // global variables and messages
 #include "errors.h"	    // error messages
